cardinal/server/types/provider.go

```diff
@@ -2,20 +2,32 @@ package types
 
 import (
 	"pkg.world.dev/world-engine/cardinal/gamestate"
-	"pkg.world.dev/world-engine/cardinal/search"
-	"pkg.world.dev/world-engine/cardinal/search/filter"
+	"pkg.world.dev/world-engine/cardinal/receipt"
 	"pkg.world.dev/world-engine/cardinal/types"
-	"pkg.world.dev/world-engine/cardinal/types/engine"
 	"pkg.world.dev/world-engine/sign"
 )
 
-type Provider interface {
+type ProviderWorld interface {
 	UseNonce(signerAddress string, nonce uint64) error
 	GetSignerForPersonaTag(personaTag string, tick uint64) (addr string, err error)
 	AddTransaction(id types.MessageID, v any, sig *sign.Transaction) (uint64, types.TxHash)
 	Namespace() string
 	GetComponentByName(name string) (types.ComponentMetadata, error)
-	Search(filter filter.ComponentFilter) search.EntitySearch
 	StoreReader() gamestate.Reader
-	GetReadOnlyCtx() engine.Context
+	HandleQuery(name string, bz []byte) ([]byte, error)
+	CurrentTick() uint64
+	ReceiptHistorySize() uint64
+	GetTransactionReceiptsForTick(tick uint64) ([]receipt.Receipt, error)
+	EvaluateCQL(cql string) ([]types.EntityStateElement, error, error)
+	GetDebugState() (types.EntityStateResponse, error)
+	BuildQueryFields() []types.FieldDetail
+}
+
+// GetWorldResponse is a type representing the json super structure that contains
+// all info about the world.
+type GetWorldResponse struct {
```

Comment:
What's the point of this being here instead of just being in the handler file?

Reply:
no point. It must've been required to be here at one point, but with all the dependencies shifting around the path is open to put it into server.

Reply:
No wait a minute. This looks appropriate. It's under `server.types` I can just put it under it's own file or something if you want to segregate it away from provider stuff. But the package looks organizationally appropriate to me. 

Why do you want this under server.handler?  @smsunarto

Reply:
You are not supposed to put types in its own package unless necessary (when it's used by multiple packages) in Go [](edited-with-Axolo)

Reply:
Ok, got it I'll move it to handler. [](edited-with-Axolo)